lib/faraday/utils.rb

```diff
@@ -95,7 +95,14 @@ def to_hash() ::Hash.new.update(self) end
 
       def parse(header_string)
         return unless header_string && !header_string.empty?
-        header_string.split(/\r\n/).
+
+        headers = header_string.split(/\r\n/)
+
+        # Find the last set of response headers.
+        start_index = headers.rindex { |x| x.match(/^HTTP\//) }
```

Comment:
According to this: https://ruby-doc.org/core-2.3.0/Array.html#method-i-rindex
I believe the best (or clearest, at least) implementation would then be:
```ruby
start_index = headers.rindex { |x| x.match(/^HTTP\//) } || 0
```